[Docs] [manage] moved Manual import section to Manage. (#13665)
yugabyte/yugabyte-db: docs/content/preview/manage/data-migration/bulk-export-ysql.md

---
title: Bulk export
headerTitle: Bulk export for YSQL
linkTitle: Bulk export
description: Bulk export for YSQL using ysql_dump.
menu:
  preview:
    identifier: manage-bulk-export-ysql
    parent: manage-bulk-import-export
    weight: 704
type: docs
---

<ul class="nav nav-tabs-alt nav-tabs-yb">
   <li >
    <a href="../bulk-export-ysql/" class="nav-link active">
      <i class="icon-postgres" aria-hidden="true"></i>
      YSQL
    </a>
  </li>
  <li >
    <a href="../bulk-export-ycql/" class="nav-link">
      <i class="icon-cassandra" aria-hidden="true"></i>
      YCQL
    </a>
  </li>
</ul>

This page describes the following steps required to export PostgreSQL data to YugabyteDB.

- [Convert a PostgreSQL schema](#convert-a-postgresql-schema)

> - [Migrate a PostgreSQL application](#migrate-a-postgresql-application/)

Reviewer: ```suggestion
- [Migrate a PostgreSQL application](#migrate-a-postgresql-application)
```

- [Export PostgreSQL data](#export-postgresql-data)

## Convert a PostgreSQL schema

To convert the PostgreSQL schema to YugabyteDB schema, the following changes need to be made.

{{< tip title="Tip" >}}

> Using `ysql_dump` tool can simplify some steps of your schema migration, read more at [use ysql_dump](#use-ysql-dump).

Reviewer: ```suggestion
The `ysql_dump` tool can simplify some steps of your schema migration, refer to [use ysql_dump](#use-ysql-dump).
```

)
TO '<table_name>_1.csv'
WITH (FORMAT CSV DELIMITER ',' HEADER);
```

```sql
COPY (
    SELECT * FROM <table_name>
    ORDER BY <primary_key_col>
    LIMIT num_rows_per_export OFFSET num_rows_per_export
)
TO '<table_name>_2.csv'
WITH (FORMAT CSV DELIMITER ',' HEADER);
```

```sql
COPY (
    SELECT * FROM <table_name>
    ORDER BY <primary_key_col>
    LIMIT num_rows_per_export OFFSET num_rows_per_export * 2
)
TO '<table_name>_3.csv'
WITH (FORMAT CSV DELIMITER ',' HEADER);
```

You can run the above commands in parallel to speed up the process. This approach will also produce multiple CSV files, allowing for parallel import on the YugabyteDB side.

### Export data into SQL script using ysql_dump

An alternative way to export the data is using the YugabyteDB [`ysql_dump`](../../../admin/ysql-dump/) backup utility, which is derived from PostgreSQL pg_dump.

```sh
$ ysql_dump -d <database_name> > <database_name>.sql
```

`ysql_dump` is the ideal option for smaller datasets, because it allows you to export a whole database by running a single command. However, the COPY command is recommended for large databases, because it significantly enhances the performance.

## Next steps

> - [Bulk export for YSQL](.../../../../data-migration/bulk-export-ysql/)

Reviewer: this is bulk export; link looks wrong as well, the file is in the same directory

Author: oops , will correct this one!
